app/models/member.rb

```diff
@@ -196,6 +196,10 @@ def interesting?
     return false
   end
 
+  def Member.login_name_or_email(login)
```

Comment:
I started with scopes - but the railsguide on how to do scopes says a class method is preferrable to a scope with a variable pass in. 
http://guides.rubyonrails.org/active_record_querying.html#passing-in-arguments

> However, this is just duplicating the functionality that would be provided to you by a class method.
> ...
> Using a class method is the preferred way to accept arguments for scopes.